x-pack/plugins/lens/public/lens_suggestions_api.ts

```diff
@@ -76,7 +80,7 @@ export const suggestionsApi = ({
     activeVisualization: visualizationMap[activeVisualization.visualizationId],
     visualizationState: activeVisualization.visualizationState,
     dataViews,
-  });
+  }).filter((sug) => !sug.hide && sug.visualizationId !== 'lnsLegacyMetric');
```

Comment:
ℹ️ Moved the filtering from Discover to Lens api, Lens won't return the hidden suggestions and the lnsLegacyMetric as it is a legacy viz.